Feature/kdt feature

kdt2014 opened the pull request:
add socks proxy function for users who need it as the origin code can not use sockts prxoy which is the dominant way.

add batch_proxy function for users who need to process a lot of srt files

<img width="898" alt="proxy" src="https://github.com/machinewrapped/gpt-subtrans/assets/86955983/c0c9c443-341b-46b5-b515-cdf12055688e">

machinewrapped reviewed (commented):
I don't know need to use a proxy so I don't know much about configuring it, but looks good. I guess `.idea` should be added to `.gitignore` so it isn't included in the repository?

kdt2014 commented:
> I don't know need to use a proxy so I don't know much about configuring it, but looks good. I guess `.idea` should be added to `.gitignore` so it isn't included in the repository?

Hi, nice to hear from you！ I added the .idea to gitignore and updated the reposity, now there is no .idea folder on github reposity.
Proxy is very useful in China, the only way they can get access to gpts, such as chatgpt, claude, gemini, is by using proxy.

Currently, the dominant proxy ways are v2ray, naive proxy and clash.
Below is the config file for v2ray(this project has 27.8k stars, which is widely used)

https://github.com/v2fly/v2ray-core/blob/master/release/config/config.json

the inbound protocol is socks. 

here is the naiveproxy (6.3k stars)

https://github.com/klzgrad/naiveproxy

"{
  "listen": "socks://127.0.0.1:1080",
  "proxy": "https://user:pass@example.com"
}
"
it also use socks as the default protocol.

If you have any questions, please let me know.

machinewrapped commented:
Sounds good - it's merged 👍
